syntax = "proto3";

package roboapp;

import "buf/validate/validate.proto";

message RobotStateMessage {
  int32 state = 1 [(buf.validate.field).required = true, (buf.validate.field).int32.gte = 0];
  string color = 2 [
    (buf.validate.field).required = true,
    (buf.validate.field).string = {
      in: [
        "blue",
        "red"
      ]
    }
  ]; // "blue" or "red"
}
